designs/NuGet-ClientCertificates.md

```diff
@@ -0,0 +1,324 @@
+
+- `-Path` `string` - Path to certificate file added to a file client certificate source.
+
+- `-Password` `string` - Password for the certificate, if needed. This option can be used to specify the password for the certificate. Available only for from file source types.
+
+- `-StorePasswordInClearText` `string` - Enables storing password for the certificate by disabling password encryption. Default value: false
+
+- `-StoreLocation` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storelocation?view=netframework-4.8) - StoreLocation added to a storage client certificate source. Default value: CurrentUser
+
+- `-StoreName` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storename?view=netframework-4.8) - StoreName added to a storage client certificate source. Default value: My
+
+- `-FindBy` [possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509findtype?view=netframework-4.8) - FindBy added to a storage client certificate source. Default value: Thumbprint
+
+- `-FindValue` `string` - FindValue added to a storage client certificate source.
+
+`-Path` and `-Password` options determine final certificate source type as `fileCert`
+
+`-StoreLocation`, `-StoreName`, `-FindBy` and `-FindValue` options determine final certificate source type as `storeCert`
+
+It is denied to use options from different certificate source type at the same time.
+
+Command will fail if user tries to change initial certificate source type.
+
+### nuget client-certificates remove -PackageSource <name>
+
+Removes any client certificate configuration that match the given package source name.
+
+### Examples
+
+```
+nuget client-certificates Add -PackageSource Foo -Path .\MyCertificate.pfx
+
+nuget client-certificates Add -PackageSource Contoso -Path c:\MyCertificate.pfx -Password 42
+
+nuget client-certificates Add -PackageSource Foo -FindValue ca4e7b265780fc87f3cb90b6b89c54bf4341e755
+
+nuget client-certificates Add -PackageSource Contoso -StoreLocation LocalMachine -StoreName My -FindBy Thumbprint -FindValue ca4e7b265780fc87f3cb90b6b89c54bf4341e755
+
+nuget client-certificates Update -PackageSource Foo -FindValue ca4e7b265780fc87f3cb90b6b89c54bf4341e755
+
+nuget client-certificates Remove -PackageSource certificateName
+
+nuget client-certificates
+
+nuget client-certificates List -Name containsInPackageSourceName
+
+nuget client-certificates List -SourceType storage
+```
+
+# dotnet tool
+
+## dotnet nuget <command> client-cert
+
+Bunch of commands similar to NuGet.exe client-certificates. 
+
+Gets, updates, sets or lists client certificates to the NuGet configuration.
+
+Usage
+```
+dotnet nuget <list|add|remove|update> client-cert [options]
+```
+### dotnet nuget list client-cert [options]
```

Comment:
Like the `nuget client-certs list` comment, I don't think the filtering options should be added.

Reply:
Done